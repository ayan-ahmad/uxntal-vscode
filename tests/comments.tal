x ( x ) x
x (x x ) x
x ( x )x ) x
x (x) x )x()() x ) x
x (x) x)X )x()() x ) x
(
    BRK LIT2kr @label/label &sublabel
    .label #12 #abcd $1 |100 ~lib.tal
    "string %MACRO { } [ ]
)

X( x )X ) x

( Nested comments )
x ( x ( x ) x ) x
x (x x ( x ) x ) x
x ( x ( x )x x ) x ) x
x (x x ( x )x x ) x ) x

( Not nested -- inner comment doesn't start with a `(` )
x ( x (x x ) x ) x
x (x x (x x ) x ) x
