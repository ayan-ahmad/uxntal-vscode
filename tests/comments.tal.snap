>x ( x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^ source.tal comment.uxntal
#       ^ source.tal
#        ^ source.tal variable.uxntal
>x (x x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^^ source.tal comment.uxntal
#    ^ source.tal
#     ^ source.tal comment.uxntal
#      ^ source.tal
#       ^ source.tal comment.uxntal
#        ^ source.tal
#         ^ source.tal variable.uxntal
>x ( x )x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^^ source.tal comment.uxntal
#        ^ source.tal
#         ^ source.tal comment.uxntal
#          ^ source.tal
#           ^ source.tal variable.uxntal
>x (x) x )x()() x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^^^ source.tal comment.uxntal
#     ^ source.tal
#      ^ source.tal comment.uxntal
#       ^ source.tal
#        ^^^^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal variable.uxntal
>x (x) x)X )x()() x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^^^ source.tal comment.uxntal
#     ^ source.tal
#      ^^^ source.tal comment.uxntal
#         ^ source.tal
#          ^^^^^^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal variable.uxntal
>(
#^ source.tal comment.uxntal
>    BRK LIT2kr @label/label &sublabel
#^^^^ source.tal
#    ^^^ source.tal comment.uxntal
#       ^ source.tal
#        ^^^^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^^^^^^^^^^^^ source.tal comment.uxntal
#                           ^ source.tal
#                            ^^^^^^^^^ source.tal comment.uxntal
>    .label #12 #abcd $1 |100 ~lib.tal
#^^^^ source.tal
#    ^^^^^^ source.tal comment.uxntal
#          ^ source.tal
#           ^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^^^^^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^^ source.tal comment.uxntal
#                       ^ source.tal
#                        ^^^^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^^^^^^^^ source.tal comment.uxntal
>    "string %MACRO { } [ ]
#^^^^ source.tal
#    ^^^^^^^ source.tal comment.uxntal
#           ^ source.tal
#            ^^^^^^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal comment.uxntal
#                        ^ source.tal
#                         ^ source.tal comment.uxntal
>)
#^ source.tal comment.uxntal
>
>X( x )X ) x
#^^ source.tal variable.uxntal
#  ^ source.tal
#   ^ source.tal variable.uxntal
#    ^ source.tal
#     ^^ source.tal variable.uxntal
#       ^ source.tal
#        ^ source.tal variable.uxntal
#         ^ source.tal
#          ^ source.tal variable.uxntal
>
>( Nested comments )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^ source.tal comment.uxntal
#        ^ source.tal
#         ^^^^^^^^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
>x ( x ( x ) x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^ source.tal comment.uxntal
#       ^ source.tal
#        ^ source.tal comment.uxntal
#         ^ source.tal
#          ^ source.tal comment.uxntal
#           ^ source.tal
#            ^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal variable.uxntal
>x (x x ( x ) x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^^ source.tal comment.uxntal
#    ^ source.tal
#     ^ source.tal comment.uxntal
#      ^ source.tal
#       ^ source.tal comment.uxntal
#        ^ source.tal
#         ^ source.tal comment.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal variable.uxntal
>x ( x ( x )x x ) x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^ source.tal comment.uxntal
#       ^ source.tal
#        ^ source.tal comment.uxntal
#         ^ source.tal
#          ^^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal variable.uxntal
>x (x x ( x )x x ) x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^^ source.tal comment.uxntal
#    ^ source.tal
#     ^ source.tal comment.uxntal
#      ^ source.tal
#       ^ source.tal comment.uxntal
#        ^ source.tal
#         ^ source.tal comment.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^ source.tal variable.uxntal
>
>( Not nested -- inner comment doesn't start with a `(` )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^ source.tal comment.uxntal
#     ^ source.tal
#      ^^^^^^ source.tal comment.uxntal
#            ^ source.tal
#             ^^ source.tal comment.uxntal
#               ^ source.tal
#                ^^^^^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^^^^^^^ source.tal comment.uxntal
#                             ^ source.tal
#                              ^^^^^^^ source.tal comment.uxntal
#                                     ^ source.tal
#                                      ^^^^^ source.tal comment.uxntal
#                                           ^ source.tal
#                                            ^^^^ source.tal comment.uxntal
#                                                ^ source.tal
#                                                 ^ source.tal comment.uxntal
#                                                  ^ source.tal
#                                                   ^^^ source.tal comment.uxntal
#                                                      ^ source.tal
#                                                       ^ source.tal comment.uxntal
>x ( x (x x ) x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^^ source.tal comment.uxntal
#        ^ source.tal
#         ^ source.tal comment.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal variable.uxntal
#              ^ source.tal
#               ^ source.tal variable.uxntal
#                ^ source.tal
#                 ^ source.tal variable.uxntal
>x (x x (x x ) x ) x
#^ source.tal variable.uxntal
# ^ source.tal
#  ^^ source.tal comment.uxntal
#    ^ source.tal
#     ^ source.tal comment.uxntal
#      ^ source.tal
#       ^^ source.tal comment.uxntal
#         ^ source.tal
#          ^ source.tal comment.uxntal
#           ^ source.tal
#            ^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal variable.uxntal
#               ^ source.tal
#                ^ source.tal variable.uxntal
#                 ^ source.tal
#                  ^ source.tal variable.uxntal
>